302 StepProgress component

PatrickDeVries opened the pull request:
Added the StepProgress component from #302. I have not written tests before so I copied the RangeSlider tests mostly, let me know if they need anything else added.

PatrickDeVries committed:
set up core stepProgress

PatrickDeVries committed:
Added labels and slider fill

PatrickDeVries committed:
circular buttons added

PatrickDeVries committed:
Added click controls, analytics, and label locations

PatrickDeVries committed:
Exposed and improved containers

PatrickDeVries committed:
Deleted commented code

PatrickDeVries committed:
Everything is square, need to bottom align over-text

PatrickDeVries committed:
added tests and refs

PatrickDeVries committed:
Merge branch 'master' into 302-stepprogress-component

PatrickDeVries committed:
Updated step progress imports

PatrickDeVries committed:
Update StepProgress.tsx

PatrickDeVries committed:
Removed empty types file

PatrickDeVries committed:
Changed default arguments to inner labels and enabled clicks

PatrickDeVries committed:
added labelType enum to index

PatrickDeVries committed:
Removed 100% width / height on container

PatrickDeVries committed:
Update StepProgress.test.tsx.snap

dawsonbooth reviewed (commented):
Should they be this big by default?

![image](https://user-images.githubusercontent.com/10625031/136592341-6e77492a-d4e6-478a-8be7-e56cd6427a1c.png)

PatrickDeVries committed:
fixed exports, width, and added stylable rails

PatrickDeVries committed:
Update StepProgress.test.tsx.snap

PatrickDeVries committed:
Added a vertical option

PatrickDeVries committed:
jest update

dawsonbooth committed:
Merge branch 'master' into 302-stepprogress-component

PatrickDeVries committed:
changed rotation origin

PatrickDeVries committed:
retriggering checks

PatrickDeVries committed:
replaced "label" with "step" in stepProgress to avoid confusion with label component

aVileBroker committed:
Merge branch 'master' into 302-stepprogress-component

PatrickDeVries committed:
feat(stepprogress): add StepProgress component

Add a new component for tracking progress through forms and navigating through them. Supports
styling options like rounding of buttons and vertical progression. Also supports disabling of
clicking forward or backward and setting onClicks for each step.

re #302

PatrickDeVries committed:
Added additional styling options and fixed step type selection issues

aVileBroker committed:
Merge branch 'master' into 302-stepprogress-component
